File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int [] answer = new int[2]; //가로, 세로를 담을 배열 생성
        int x, y;
        //가로 길이는 세로 길이와 같거나, 세로 길이보다 길다고 했으니 세로의 최댓값
        for (int i = 1; i <= Math.sqrt(yellow); i++) {
            //세로 길이는 +2
            y = i + 2;
            //가로 길이도 + 2
            x = (yellow/i) + 2;
            //가로 * 세로 == 전체의 합과 같다면?
            if (x * y == brown + yellow) {
                answer[0] = x;
                answer[1] = y;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
# 약수 구하기 함수
def getMyDivisor(n):

    divisorsList = []

    for i in range(1, int(n**(1/2)) + 1):
        if (n % i == 0):
            divisorsList.append(i) 
            if ( (i**2) != n) : 
                divisorsList.append(n // i)

    divisorsList.sort()
    return divisorsList
        
def solution(brown, yellow):
    answer = []
    
    # 노란색 개수로 약수 리스트 
    for i in getMyDivisor(yellow):
        
        # (노란색 가로 길이 + 2) x 2 
        # + (노란색 세로 길이) X 2 의 개수가 갈색 개수랑 똑같으면 
        # 노란색 가로 + 2 , 노란색 세로 + 2 행렬이 됨
        if( (((i+2) * 2) + (yellow // i) *2) == brown ) :
            answer.append(i+2)
            answer.append((yellow // i) + 2)
            break
    
    # 내림차순으로 정렬 해서 리턴
    answer.sort(reverse=True)

    return answer